code/_onclick/ai.dm

```diff
@@ -63,6 +66,10 @@
 		waypoint_mode = 0
 		return
 
+	if(controlled_mech) //Are we piloting a mech? Placed here so the modifiers are not overridden.
```

Comment:
does byond support function pointers? if it does, you could write two different functions, and then make onclick point to either of them accordingly. this would change at runtime in enter/exit mech functions.